### ISSUE TITLE
services: Unit not found pages have no action

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#16946]
Generally, you probably won't hit a unit not found page. But in bugs like #16945, you may. And there may be some edge cases where you could hit it, especially for user services which might be spawned just for one session and not for another.

Right now, the page looks like this:

![image](https://user-images.githubusercontent.com/10246/152994893-75bf7f7a-f3f8-4673-81f3-38c9b4139351.png)

We should use an [empty state pattern](https://www.patternfly.org/v4/components/empty-state/) with an action. The action would just be a link back to the main page. (It should be styled like a link, not a button, as it goes somewhere instead of doing something.)

---

**TL;DR**: Add a link under "Unit not found" that says "View all services" which goes back to the services list.

**Comment by Zach-Clay** [comment]
Hi! Could I be assigned to this issue? I have not contributed to this codebase before and I think this would be a good start.

**Comment by marusak** [comment]
@Zach-Clay happy to see your interest :) I have assigned this issue to you. If you have any questions I'm happy to help! Thanks!

**Comment by Zach-Clay** [comment]
Hey @marusak ! So I'm trying to get started on this issue and have a few questions for you. So I have setup up my environment on Fedora and have my forked repository cloned and ready to go. 

1. From here, what is the easiest way to run the local repo in a web server so I can track my changes? I'm trying to look through the HACKING.md guide and having a little bit of trouble with starting the local server.
2. What file/files should I be looking at for this issue?

Thank you for your time!

**Comment by marusak** [comment]
> From here, what is the easiest way to run the local repo in a web server so I can track my changes? I'm trying to look through the HACKING.md guide and having a little bit of trouble with starting the local server.

[building](https://github.com/cockpit-project/cockpit/blob/main/HACKING.md#building)  and [symlinking](https://github.com/cockpit-project/cockpit/blob/main/HACKING.md#working-on-your-local-machine-cockpits-session-pages) should be all you need to do. These four commands usually should do:

```
./autogen.sh --prefix=/usr --enable-debug
make
mkdir -p ~/.local/share/
ln -s $(pwd)/dist ~/.local/share/cockpit
```

Then I suggest changing some obvious string (like some title on overview page), rebuliding, reloading in browser and making sure you can see it.

> What file/files should I be looking at for this issue?

```
 $ git grep "Unit not found"
pkg/systemd/services/services.jsx:                            title={_("Unit not found")} />;
```

Then `EmptyStatePanel` is defined in `pkg/lib/cockpit-components-empty-state.jsx`


P.S. you are talking about `web server` and `starting the local server`... Not super sure what you mean. Cockpit is used for maintaining of some serveres - it can be your local machine - so when you just enable cockpit socket and visit `localhost:9090` you should be able to log in with you normal credentials and manage you local machine. Surely that is good enough environment for this issue.

**Comment by Zach-Clay** [comment]
@marusak When doing the building, I am running into this error:
<img width="433" alt="Screenshot 2022-02-09 170818" src="https://user-images.githubusercontent.com/75220023/153298551-80d65048-3414-44f0-9300-0fa305ac14dc.png">

**Comment by marusak** [comment]
> `autoreconf: command not found`

You need to install `autoreconf`. It should be pulled in when you prepare [development dependencies](https://github.com/cockpit-project/cockpit/blob/main/HACKING.md#getting-the-development-dependencies). Have you done these steps and it was not installed?

**Comment by Zach-Clay** [comment]
@marusak I got it :)! So sorry for all the back and forth about such a simple issue... I'll have a PR in within the next few days!
